src/bridge/bridge.c

```diff
@@ -368,9 +368,6 @@ run_bridge (const gchar *interactive,
         g_warning ("couldn't change to runtime dir: %s: %s", directory, g_strerror (errno));
     }
 
-  /* Reset the umask, typically this is done in .bashrc for a login shell */
-  umask (022);
```

Comment:
This is actually a good point. Given the test failures, it is actually *certain* that cockpit would break some things such as /etc/motd permissions if the OS uses a tight 077 umask by default. We are once again at the point where issue #18033 bites us.

So from a "clean room" perspective, I'd like to drop all this umask mucking around (in both the bridge and c-session), and make cockpit.file() respect existing file permissions, or go over all API users and explicitly specify owner and permissions (which is sometimes annoying and race-y to figure out from JS, though). But as we don't really know who else uses this API, it now seems to me that this is water under the bridge, and we just need to do that umask thing in the Py bridge as well. Would that be acceptable?